src/operator/activation.cu

```diff
@@ -14,6 +14,10 @@ namespace mxnet {
 namespace op {
 template<>
 Operator *CreateOp<gpu>(ActivationParam param) {
+  // Not supported by CUDNN yet
+  if (param.act_type == activation::kSoftplus)
+      return new ActivationOp<gpu, mshadow_op::softplus, mshadow_op::softplus_grad>();
+
```

Comment:
Are you sure?  If I remove this check a FATAL error will be thrown by the `CuDNNActivationOp` constructor.  In contrast, the current patch works with and without CUDNN support compiled.